Docs/source/running_cpp/parameters.rst

```diff
@@ -949,16 +949,104 @@ Numerics and algorithms
     In 2D simulations, only the first two values are read.
 
 * ``algo.current_deposition`` (`string`, optional)
-    The algorithm for current deposition. Available options are:
-
-     - ``esirkepov``: the charge-conserving Esirkepov algorithm
-       (see `Esirkepov, Comp. Phys. Comm. (2001) <https://www.sciencedirect.com/science/article/pii/S0010465500002289>`__)
-     - ``direct``: simpler current deposition algorithm, described in
-       the section :doc:`../theory/picsar_theory`. Note that this algorithm is not strictly charge-conserving.
-
-    If ``algo.current_deposition`` is not specified, the default is
-    ``esirkepov`` (unless WarpX is compiled with ``USE_PSATD=TRUE``, in which
-    case the default is ``direct``).
+    This parameter selects the algorithm for the deposition of the current density.
+    Available options are: ``direct``, ``esirkepov``, and ``vay``. The default choice
+    is ``esirkepov`` if WarpX is compiled with the FDTD solver (that is, with
+    ``USE_PSATD=FALSE``) and ``direct`` if WarpX is compiled with the standard or
+    Galilean PSATD solver (that is, with ``USE_PSATD=TRUE``).
+
+    1. ``direct``
+
+       The current density is deposited as described in the section :doc:`../theory/picsar_theory`.
+       This deposition scheme does not conserve charge.
+
+    2. ``esirkepov``
+
+       The current density is deposited as described in
+       `(Esirkepov, CPC, 2001) <https://www.sciencedirect.com/science/article/pii/S0010465500002289>`_.
+       This deposition scheme guarantees charge conservation for shape factors of arbitrary order.
+
+    3. ``vay``
+
+       The current density is deposited as described in
+       `(Vay et al, JCP 243, 2013) <https://doi.org/10.1016/j.jcp.2013.03.010>`_.
+       This deposition scheme is the generalization of the Esirkepov deposition scheme
+       for the spectral case with arbitrary-order stencils. The current density
+       :math:`\widehat{\boldsymbol{J}}^{\,n+1/2}` in Fourier space is computed as
+       :math:`\widehat{\boldsymbol{J}}^{\,n+1/2} = i \, \widehat{\boldsymbol{D}} / \boldsymbol{k}`
+       when :math:`\boldsymbol{k} \neq 0` and set to zero otherwise. The quantity
+       :math:`\boldsymbol{D}` is deposited in real space by averaging the currents
+       over all possible grid paths between the initial position :math:`\boldsymbol{x}^{\,n}`
+       and the final position :math:`\boldsymbol{x}^{\,n+1}` and is defined as
```

Comment:
Thanks for the very nice description!
One minor remark: could you move the equations to the document `../theory/picsar_theory` (you can create a new paragraph/section there for the Vay deposition) and point to it from this file. I think that this would make the description of the code parameters (in the current document) more streamlined.